vts/plugins/scheme-cca-ssd-platform/main.go

```diff
@@ -0,0 +1,459 @@
+	}
+
+	if len(rest) != 0 {
+		return errors.New("trailing data found after PEM block")
+	}
+
+	if block.Type != "PUBLIC KEY" {
+		return fmt.Errorf("unsupported key type %q", block.Type)
+	}
+
+	pk, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return err
+	}
+
+	var ccaToken ccatoken.Evidence
+
+	if err = ccaToken.FromCBOR(token.Data); err != nil {
+		return err
+	}
+
+	if err = ccaToken.Verify(pk); err != nil {
+		return err
+	}
+	log.Println("\n CCA platform token signature, realm token signature and cryptographic binding verified")
+	return nil
+}
+
+func (s Scheme) AppraiseEvidence(
+	ec *proto.EvidenceContext, endorsementsStrings []string,
+) (*ear.AttestationResult, error) {
+	var endorsements []Endorsements
+
+	result := ear.NewAttestationResult()
+
+	for i, e := range endorsementsStrings {
+		var endorsement Endorsements
+
+		if err := json.Unmarshal([]byte(e), &endorsement); err != nil {
+			return nil, fmt.Errorf("could not decode endorsement at index %d: %w", i, err)
+		}
+
+		endorsements = append(endorsements, endorsement)
+	}
+
+	err := populateAttestationResult(result, ec.Evidence.AsMap(), endorsements)
+
+	// TO DO: Need to populate Unprocessed Evidence in a suitable format in AR
+	/*
+		// Unprocessed evidence should be a JSON Byte Array, which can be UnMarshalled by EAR Library
+		result.unprocessed_evidence = json.Marshal(ec.UpEvidence.AsMap())
+	*/
+
+	return result, err
+}
+
+type ClaimMapper interface {
+	ToJSON() ([]byte, error)
+}
+
+func claimsToMap(mapper ClaimMapper) (map[string]interface{}, error) {
+	data, err := mapper.ToJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	var out map[string]interface{}
+	err = json.Unmarshal(data, &out)
+
+	return out, err
+}
+
+func mapToClaims(in map[string]interface{}) (psatoken.IClaims, error) {
+	data, err := json.Marshal(in)
+	if err != nil {
+		return nil, err
+	}
+
+	return psatoken.DecodeJSONClaims(data)
+}
+
+func populateAttestationResult(
+	result *ear.AttestationResult,
+	evidence map[string]interface{},
+	endorsements []Endorsements,
+) error {
+	claims, err := mapToClaims(evidence)
+	if err != nil {
+		return err
+	}
+
+	// once the signature on the token is verified, we can claim the HW is
+	// authentic
+	result.TrustVector.Hardware = ear.GenuineHardwareClaim
+
+	swComps := filterRefVal(endorsements, "cca.sw-component")
+	match := matchSoftware(claims, swComps)
+	if match {
+		result.TrustVector.Executables = ear.ApprovedRuntimeClaim
+		log.Println("\n matchSoftware Success")
+
+	} else {
+		result.TrustVector.Executables = ear.UnrecognizedRuntimeClaim
+		log.Println("\n matchSoftware Failed")
+	}
+
+	platformConfig := filterRefVal(endorsements, "cca.platform-config")
+	match = matchPlatformConfig(claims, platformConfig)
+
+	if match {
+		result.TrustVector.Configuration = ear.ApprovedConfigClaim
+		log.Println("\n matchPlatformConfig Success")
+
+	} else {
+		result.TrustVector.Configuration = ear.UnsafeConfigClaim
```

Comment:
AR4si does not have symantic

Reply:
https://github.com/ietf-rats-wg/draft-ietf-rats-ar4si/issues/17
added to address this issue in ar4si draft